### ISSUE TITLE
Investigate trimming DCMTK for builds

### ISSUE THREAD

**Issue by thewtex** [InsightSoftwareConsortium/itk-wasm#630]
Look for data embedded in the binaries that can be removed to improve load time.

Start by examining the sizes of the `.a` archives.

**Comment by michaelonken** [comment]
Yes, but it will need some time. Since I am not in the topic,  I have some questions:
- Do you refer to download time (sheer size of the DCMTK-related download) or startup time of the downloaded WASM "binary"?
- Which part of DCMTK (libraries, executables) are you currently using (maybe we can simply disable some)?
- If we have those parts we could further check whether there are features which we could disable.

**Comment by thewtex** [comment]
> Do you refer to download time (sheer size of the DCMTK-related download) or startup time of the downloaded WASM "binary"?

@michaelonken yes, the resulting wasm module's size is extremely important because of 1) download time, and 2) time for the wasm interpreter to compile it.

> Which part of DCMTK (libraries, executables) are you currently using (maybe we can simply disable some)?

We only link to the dcmtk libraries needed, currently `ofstd oflog dcmdata dcmimgle dcmimage dcmsr dcmpstat dcmnet dcmqrdb`, and there are a few more needed for SEG support.

The wasm toolchain does an amazing job at dead code elimination. We may need to help it along or refactor to make it easier, split the libraries, etc. Currently, the dcmtk-based wasm modules are much larger than the others.

Here is one related tool, I have not tried it myself. ITK-Wasm does [support debug builds](https://wasm.itk.org/en/latest/cxx/tutorial/debugging.html).

> If we have those parts we could further check whether there are features which we could disable.

More details:

https://github.com/InsightSoftwareConsortium/itk-wasm/blob/main/packages/dicom/dcmtk/CMakeLists.txt
